src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -2895,13 +2895,21 @@ module Crystal
     def visit(node : TupleIndexer)
       scope = @scope
       if scope.is_a?(TupleInstanceType)
-        node.type = scope.tuple_types[node.index].as(Type)
+        if (index = node.index).is_a?(Range)
```

Comment:
This is a great opportunity to use `case .. in Range, Int32` so that if new range types are added, the compiler will tell us (unlikely, but maybe it makes the code clearer). Same goes for the code a bit below.